File: 2-4/2-4-TS/fruits.ts
/*
 * This is a program that uses mrCoxallStack class for fruit items.
 * @author  Jakub Malhotra
 * @version 3.0 
 * @since   2024-10-21
 */

import MrCoxallStack from './mrCoxallStack.ts'

// fruit Stack
const fruitStack = new MrCoxallStack()
console.log(`Fruit items: ${fruitStack.showStack}`)
console.log(`The size of the stack is: ${fruitStack.size}`)
console.log(`Is the stack empty? ${fruitStack.isEmpty}`)
// push items
fruitStack.pushItem('banana')
fruitStack.pushItem('apple')
fruitStack.pushItem('cherry')
// print full stack
console.log(`Fruit items: ${fruitStack.showStack}`)
// pop and show an item
const topFruit = fruitStack.popItem();
console.log(`Popped fruit item: ${topFruit}`);
// display stack after popping item
console.log(`Fruit items: ${fruitStack.showStack}`)
console.log(`The size of the stack is: ${fruitStack.size}`)
console.log(`Is the stack empty? ${fruitStack.isEmpty}`)
// peek and display an item
console.log(`The top fruit is: ${fruitStack.peekItem}`)
console.log(`The size of the stack is: ${fruitStack.size}`)
// empty the stack and show it
console.log(`Emptying the stack ...`)
fruitStack.clearStack
console.log(`The size of the stack is: ${fruitStack.size}`)
console.log(`Is the stack empty? ${fruitStack.isEmpty}`)

console.log('\nDone.')

File: 2-4/2-4-Java/Fruits.java
/*
 * This is a program that uses MrCoxallStack class for fruit items.
 * @author  Jakub Malhotra
 * @version 3.0
 * @since   2024-10-21
 */

public final class Fruits {

    // Private constructor to prevent instantiation of utility class
    private Fruits() {
        throw new IllegalStateException("This is the main function");
    }

    /**
     * The main method uses MrCoxallStack for fruit items.
     *
     * @param args command line arguments
     */
    public static void main(final String[] args) {
        // Fruit Stack
        MrCoxallStack fruitStack = new MrCoxallStack();
        System.out.println("Fruit items: " + fruitStack.getShowStack());
        System.out.println("The size of the stack is: " + fruitStack.getSize());
        System.out.println("Is the stack empty? " + fruitStack.isEmpty());

        // Push items
        fruitStack.pushItem("banana");
        fruitStack.pushItem("apple");
        fruitStack.pushItem("cherry");

        // Print full stack
        System.out.println("Fruit items: " + fruitStack.getShowStack());

        // Pop and show an item
        String topFruit = fruitStack.popItem();
        System.out.println("Popped fruit item: " + topFruit);

        // Display stack after popping item
        System.out.println("Fruit items: " + fruitStack.getShowStack());
        System.out.println("The size of the stack is: " + fruitStack.getSize());
        System.out.println("Is the stack empty? " + fruitStack.isEmpty());

        // Peek and display an item
        System.out.println("The top fruit is: " + fruitStack.peekItem());
        System.out.println("The size of the stack is: " + fruitStack.getSize());

        // Empty the stack and show it
        System.out.println("Emptying the stack ...");
        fruitStack.clearStack();
        System.out.println("The size of the stack is: " + fruitStack.getSize());
        System.out.println("Is the stack empty? " + fruitStack.isEmpty());

        System.out.println("\nDone.");
    }
}
